### PR TITLE
Avoid doing I/O in repr() or str()

### PR DESCRIPTION
**Pull request by pitrou**
Also standardize on defining `__repr__`.
`__str__` automatically falls back on `__repr__` unless overriden.

### CODE REVIEW

File: distributed/tests/test_client.py [hunk 1]
```diff
@@ -1694,7 +1694,7 @@ def test_repr(loop):
             for func in funcs:
                 text = func(c)
                 assert c.scheduler.address in text
-                assert '2' in text
```
Comment (pitrou) [inline distributed/tests/test_client.py]: This would succeed simply because the address of the scheduler is always `tcp://127.0.0.1:something` ;-)

**Comment by pitrou** [comment]
@mrocklin I'd like to merge this quickly.

**Comment by mrocklin** [comment]
I often use the repr to check how many cores/RAM I have in the cluster.  This can change during the course of a client's lifetime.  It is valuable to check in with the scheduler in the repr.

**Comment by pitrou** [comment]
Well, there is a periodic callback to update that information every 2 seconds. That should be good enough?

**Comment by pitrou** [comment]
We can't do I/O in `__repr__` as it can be invoked at any time. For example, asyncio in debug mode invokes `repr()` on callback arguments... if you do I/O by calling back into asyncio at that point, things completely break.

**Comment by mrocklin** [comment]
Ah I missed that, yes, that should be good enough.

**Comment by pitrou** [comment]
Cool, thanks :)

**Pull request merged**